### PR TITLE
refactor: strict null checks for util

### PR DESCRIPTION
**Pull request by viceice**
## Changes

- make util null safe

## Context
- #7154 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: tsconfig.strict.json [hunk 1]
```diff
@@ -266,13 +266,7 @@
...
     "lib/modules/platform/index.ts",
     "lib/renovate.ts",
-    "lib/util/exec/buildpack.ts",
-    "lib/util/exec/docker/index.ts",
-    "lib/util/exec/index.ts",
-    "lib/util/git/index.ts",
-    "lib/util/git/private-key.ts",
-    "lib/util/git/url.ts",
-    "lib/util/package-rules.ts",
+    "lib/util/package-rules.ts", // depends on versionings
```
Comment (viceice) [inline tsconfig.strict.json:269]:
```suggestion
    "lib/util/package-rules.ts", // depends on config -> manager
```

**Review by JamieMagee — DISMISSED** [review]
Approved, but disabled automerge because it looks like you're still pushing commits.

File: lib/util/package-rules.ts [hunk 2]
```diff
@@ -114,21 +114,21 @@ function matchesRule(
...
     const isMatch = matchDatasources.includes(datasource);
     if (!isMatch) {
       return false;
     }
     positiveMatch = true;
   }
-  if (matchUpdateTypes.length) {
+  if (matchUpdateTypes.length && updateType) {
     const isMatch =
       matchUpdateTypes.includes(updateType) ||
```
Comment (viceice) [inline lib/util/package-rules.ts:133]:
```suggestion
  if (matchUpdateTypes.length) {
    const isMatch =
      (updateType && matchUpdateTypes.includes(updateType)) ||
```

File: lib/util/package-rules.ts [hunk 3]
```diff
@@ -221,7 +221,7 @@ function matchesRule(
     }
     positiveMatch = true;
   }
-  if (matchCurrentVersion) {
+  if (matchCurrentVersion && versioning && currentValue) {
```
Comment (viceice) [inline lib/util/package-rules.ts:224]:
```suggestion
  if (matchCurrentVersion && currentValue) {
```

**Comment by viceice** [comment]
> Approved, but disabled automerge because it looks like you're still pushing commits.

Yes, sorry. failed tests and coverage. 🙈

**Comment by viceice** [comment]
runned full suite locally, so tests and coverage should be really fixed now 🙈

**Review by rarkins — APPROVED** [review]

**Pull request merged**